src/imgui/ImGuiSettings.cc

```diff
@@ -85,6 +85,81 @@ void ImGuiSettings::setStyle()
 	}
 	setColors(selectedStyle);
 }
+
+bool ImGuiSettings::shortcutAction(ShortcutIndex index)
+{
+	static constexpr auto mods = std::array{
+		ImGuiKey_LeftCtrl, ImGuiKey_LeftShift, ImGuiKey_LeftAlt, ImGuiKey_LeftSuper,
+		ImGuiKey_RightCtrl, ImGuiKey_RightShift, ImGuiKey_RightAlt, ImGuiKey_RightSuper,
+		ImGuiKey_ReservedForModCtrl, ImGuiKey_ReservedForModShift, ImGuiKey_ReservedForModAlt,
+		ImGuiKey_ReservedForModSuper, ImGuiKey_MouseLeft, ImGuiKey_MouseRight, ImGuiKey_MouseMiddle,
+		ImGuiKey_MouseX1, ImGuiKey_MouseX2, ImGuiKey_MouseWheelX, ImGuiKey_MouseWheelY,
+	};
+	ImGuiIO& io = ImGui::GetIO();	
+	ImGuiKeyChord keyChord = ImGuiKey_None;
+	for (int key = ImGuiKey_NamedKey_BEGIN; key < ImGuiKey_NamedKey_END; ++key ) {
+		if (contains(mods, key)) continue; // skip: mods can't be primary keys in a KeyChord
+		if (ImGui::IsKeyPressed((ImGuiKey) key)) {
+			keyChord = key | (io.KeyCtrl ? ImGuiMod_Ctrl : 0) | (io.KeyShift ? ImGuiMod_Shift : 0)
+				| (io.KeyAlt ? ImGuiMod_Alt : 0) | (io.KeySuper ? ImGuiMod_Super : 0);
+			break;
+		}
+	}
+	if (keyChord != ImGuiKey_None) {
+		manager.getShortcuts().setShortcut(index, keyChord);
+		return true;
+	}
+	return false;
+}
+
+void ImGuiSettings::paintShortcutTableItem(ShortcutIndex index)
+{
+	auto shortcutName = Shortcuts::getShortcutName(index);
+	auto &shortcut = manager.getShortcuts().getShortcut(index);
+	auto popupName = strCat("shortcutModal##", shortcutName);
+
+	if (ImGui::TableNextColumn()) {
+		ImGui::Checkbox(strCat("local##", shortcutName).c_str(), &shortcut.local);
+	}
+	if (ImGui::TableNextColumn()) {
+		ImGui::Checkbox(strCat("repeat##", shortcutName).c_str(), &shortcut.repeat);
+	}
+	if (ImGui::TableNextColumn()) {
+		if (auto label = getKeyChordName(manager.getShortcuts().getShortcut(index).keyChord)) {
```

Comment:
Detail: 10 lines higher you've already stored `manager.getShortcuts().getShortcut(index)` in `shortcut`. Better reuse that variable here.

Also the subexpression `manager.getShortcuts()` is repeated several times in the code below. Maybe also factor that out.